### ISSUE TITLE
BOM parsing issue

### ISSUE THREAD

**Issue by baderdean** [yifeikong/curl_cffi#122]
Hello, 

**Expected**
Some utf-8 files begin with a BOM (https://stackoverflow.com/questions/50130605/python-2-7-csv-file-read-write-xef-xbb-xbf-code) like the swebpage https://copytop.com/. 

**Problem**
curl-cffi seems unable to decode it.

```
  File "/home/bader/code/python3/lib/python3.11/site-packages/curl_cffi/requests/cookies.py", line 47, in text
    return self.content.decode(self.charset)
           │    │       │      │    └ 'utf-8'
           │    │       │      └ <curl_cffi.requests.cookies.Response object at 0x7f1885012790>
           │    │       └ <method 'decode' of 'bytes' objects>
           │    └ b'\xef\xbb\xbf<!DOCTYPE html>\r\n<html lang="fr">\r\n   <head>\r\n      <!--[if IE]>\r\n      <meta http-equiv="X-UA-Compatib...
           └ <curl_cffi.requests.cookies.Response object at 0x7f1885012790>

Exception: 'utf-8' codec can't decode byte 0xe8 in position 115419: invalid continuation byte

```

**Comment by yifeikong** [comment]
Hmm, suprised to see there are web pages with BOM. Anyway, you can use your own decoding.

    text = rsp.content.decode("utf-8-sig")

    # or delete it
    text = rsp.content.replace(b"\xef\xbb\xbf", b"").decode()

**Comment by yifeikong** [comment]
Fixed in 0.5.10b3
